agent/src/agent_manager.rs

```diff
@@ -139,38 +141,39 @@ impl AgentManager {
         }
     }
 
-    async fn store_and_forward_own_workload_states(&mut self, to_server_msg: ToServer) {
-        log::debug!("Storing and forwarding own workload states.");
-
-        let ToServer::UpdateWorkloadState(common::commands::UpdateWorkloadState {
-            workload_states,
-        }) = to_server_msg
-        else {
-            std::unreachable!("expected UpdateWorkloadState msg.");
-        };
+    async fn store_and_forward_own_workload_states(
+        &mut self,
+        mut new_workload_state: WorkloadState,
+    ) {
+        // execute hysteresis on the local workload states as we could be stopping
+        // [impl->swdd~agent-manager-hysteresis_on-workload-states-of-its-workloads~1]
+        if let Some(old_execution_state) = self
+            .workload_state_store
+            .get_state_of_workload(new_workload_state.instance_name.workload_name())
+        {
+            // TODO: fix this shit
+            new_workload_state.execution_state = old_execution_state
+                .clone()
+                .transition(new_workload_state.execution_state);
+        }
```

Comment:
Shall this be fixed in this PR, or later?
Suggestion:
Introduce something like: my_workload_state.transition(old_execution_state).

Reply:
Fixed.